backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/modelvalidator/C_Order.java

```diff
@@ -139,4 +171,128 @@ else if (X_C_DocType.DOCSUBTYPE_PrepayOrder.equals(dt.getDocSubType())
+			}
+			if (use5_orderToIc && !Objects.equals(invoiceCandidate.getUserElementString5(), order.getUserElementString5()))
+			{
+				invoiceCandidate.setUserElementString5(order.getUserElementString5());
+			}
+			if (use6_orderToIc && !Objects.equals(invoiceCandidate.getUserElementString6(), order.getUserElementString6()))
+			{
+				invoiceCandidate.setUserElementString6(order.getUserElementString6());
+			}
+			if (use7_orderToIc && !Objects.equals(invoiceCandidate.getUserElementString7(), order.getUserElementString7()))
+			{
+				invoiceCandidate.setUserElementString7(order.getUserElementString7());
+			}
+
+			invoiceCandDAO.save(invoiceCandidate);
+		}
+	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_AFTER_CHANGE },
+			ifColumnsChanged = { I_C_Order.COLUMNNAME_UserElementString1,
+					I_C_Order.COLUMNNAME_UserElementString2,
+					I_C_Order.COLUMNNAME_UserElementString3,
+					I_C_Order.COLUMNNAME_UserElementString4,
+					I_C_Order.COLUMNNAME_UserElementString5,
+					I_C_Order.COLUMNNAME_UserElementString6,
+					I_C_Order.COLUMNNAME_UserElementString7 }
+
+	)
+	public void updateReversedInvoicesUserElementStrings(@NonNull final I_C_Order order)
+	{
+		final ClientAndOrgId clientAndOrgId = ClientAndOrgId.ofClientAndOrg(order.getAD_Client_ID(), order.getAD_Org_ID());
+
+		final boolean use1_orderToReversedInvoice = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement1_DirectUpdateOrderToInvoice_IfReversed, false, clientAndOrgId);
+		final boolean use2_orderToReversedInvoice = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement2_DirectUpdateOrderToInvoice_IfReversed, false, clientAndOrgId);
+		final boolean use3_orderToReversedInvoice = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement3_DirectUpdateOrderToInvoice_IfReversed, false, clientAndOrgId);
+		final boolean use4_orderToReversedInvoice = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement4_DirectUpdateOrderToInvoice_IfReversed, false, clientAndOrgId);
+		final boolean use5_orderToReversedInvoice = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement5_DirectUpdateOrderToInvoice_IfReversed, false, clientAndOrgId);
+		final boolean use6_orderToReversedInvoice = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement6_DirectUpdateOrderToInvoice_IfReversed, false, clientAndOrgId);
+		final boolean use7_orderToReversedInvoice = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement7_DirectUpdateOrderToInvoice_IfReversed, false, clientAndOrgId);
+
+		final List<I_C_Invoice> invoices = invoiceDAO.getInvoicesForOrderIds(Collections.singletonList(OrderId.ofRepoId(order.getC_Order_ID())));
+
+		for (final I_C_Invoice invoice : invoices)
+		{
+			final DocStatus docStatus = DocStatus.ofNullableCodeOrUnknown(invoice.getDocStatus());
+			if (!docStatus.isReversed())
+			{
+				continue;
+			}
+
+			if (use1_orderToReversedInvoice && !Objects.equals(invoice.getUserElementString1(), order.getUserElementString1()))
```

Comment:
IMHO you don't need `&& !Objects.equals(invoice.getUserElementString1(), order.getUserElementString1())`